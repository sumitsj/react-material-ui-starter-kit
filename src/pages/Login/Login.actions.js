import { createAction } from '../../utils';

export const ACTION_TYPES = {
  LOGIN_REQUESTED: 'LOGIN_REQUESTED',
  LOGIN_SUCCEEDED: 'LOGIN_SUCCEEDED',
  LOGIN_FAILED: 'LOGIN_FAILED',
};

export const login = createAction(ACTION_TYPES.LOGIN_REQUESTED);
export const loginSuccess = createAction(ACTION_TYPES.LOGIN_SUCCEEDED);
export const loginFailure = createAction(ACTION_TYPES.LOGIN_FAILED);
