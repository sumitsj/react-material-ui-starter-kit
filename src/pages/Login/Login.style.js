import makeStyles from '@material-ui/core/styles/makeStyles';

const useStyles = makeStyles((theme) => ({
  header: {
    padding: theme.spacing(8),
    [theme.breakpoints.down('sm')]: {
      padding: theme.spacing(4),
    },
    backgroundColor: theme.palette.primary.main,
  },
  divider: {
    marginLeft: theme.spacing(4),
    marginRight: theme.spacing(4),
    [theme.breakpoints.down('sm')]: {
      marginLeft: theme.spacing(2),
      marginRight: theme.spacing(2),
    },
    backgroundColor: theme.palette.secondary.main,
  },
  title: {
    maxWidth: 510,
  },
  paper: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    paddingTop: theme.spacing(4),
  },
  avatar: {
    margin: theme.spacing(1),
    backgroundColor: theme.palette.secondary.main,
  },
  form: {
    width: '100%', // Fix IE 11 issue.
    marginTop: theme.spacing(1),
  },
  submit: {
    lineHeight: '1rem',
    letterSpacing: '1.25px',
    padding: theme.spacing(2.5, 1.75),
    margin: theme.spacing(5.25, 0, 4),
  },
  link: {
    display: 'block',
    fontWeight: 500,
    lineHeight: '1rem',
    letterSpacing: '1.25px',
    textAlign: 'center',
    width: '100%',
  },
}));

export default useStyles;
