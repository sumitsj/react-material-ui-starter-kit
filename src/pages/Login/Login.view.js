import React from 'react';
import Typography from '@material-ui/core/Typography';

const Login = () => (
  <Typography variant="h6" style={{ backgroundColor: 'cyan' }}>
    Login Page
  </Typography>
);

export default Login;
