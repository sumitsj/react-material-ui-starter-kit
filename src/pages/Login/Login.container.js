import React from 'react';
import { Redirect } from 'react-router-dom';
import Login from './Login.view';

const LoginContainer = () => {
  const isAuthenticated = localStorage.getItem('auth-token');
  const [username, setUsername] = React.useState('');
  const [password, setPassword] = React.useState('');
  const [showPassword, setShowPassword] = React.useState(false);

  const props = {
    username,
    password,
    showPassword,
    setUsername,
    setPassword,
    setShowPassword,
  };

  return (
    <>
      {isAuthenticated && <Redirect to="/" />}
      <Login {...props} />
    </>
  );
};

export default LoginContainer;
