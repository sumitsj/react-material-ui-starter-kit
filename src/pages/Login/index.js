import Login from './Login.view';

export default Login;
