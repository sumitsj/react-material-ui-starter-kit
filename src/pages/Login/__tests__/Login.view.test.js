import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import Login from '../Login.view';

jest.mock('@material-ui/core/Link', () => jest.fn(() => <forget-password />));

describe('Login', () => {
  const props = {
    username: '8888888888',
    password: 'Abcd@123',
    showPassword: true,
    setUsername: jest.fn(),
    setPassword: jest.fn(),
    setShowPassword: jest.fn(),
  };

  const setItemSpy = jest.spyOn(localStorage, 'setItem');

  beforeEach(() => {
    setItemSpy.mockReset();
  });

  describe('HTML Structure', () => {
    it('should render properly when showPassword is true', () => {
      const { asFragment } = render(<Login {...props} />);
      expect(asFragment()).toMatchSnapshot();
    });

    it('should render properly when showPassword is false', () => {
      const { asFragment } = render(<Login {...props} showPassword={false} />);
      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('Events', () => {
    it('should call setUsername on username change', () => {
      const { container } = render(<Login {...props} />);
      fireEvent.change(container.querySelector('[name="username"]'), { target: { value: '1234567890' } });
      expect(props.setUsername).toHaveBeenCalledWith('1234567890');
    });

    it('should call setPassword on password change', () => {
      const { container } = render(<Login {...props} />);
      fireEvent.change(container.querySelector('[name="password"]'), { target: { value: 'abcd@1234' } });
      expect(props.setPassword).toHaveBeenCalledWith('abcd@1234');
    });

    it('should call setShowPassword on click of password visibility toggle', () => {
      const { getByLabelText } = render(<Login {...props} />);
      fireEvent.click(getByLabelText('toggle password visibility'));
      expect(props.setShowPassword).toHaveBeenCalledWith(false);
    });

    it('should call setItem on sign in click', () => {
      const { getByRole } = render(<Login {...props} />);
      fireEvent.click(getByRole('button', { name: 'Sign In' }));
      expect(localStorage.setItem).toHaveBeenCalledWith('auth-token', 'true');
    });
  });
});
