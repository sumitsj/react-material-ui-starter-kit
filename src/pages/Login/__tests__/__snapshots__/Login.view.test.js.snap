// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login HTML Structure should render properly when showPassword is false 1`] = `
<DocumentFragment>
  <main
    class="MuiContainer-root MuiContainer-maxWidthXs"
  >
    <div
      class="makeStyles-paper-291"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-292 MuiAvatar-colorDefault"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiAvatar-fallback"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
          />
        </svg>
      </div>
      <h1
        class="MuiTypography-root MuiTypography-h5"
      >
        Sign in
      </h1>
      <form
        class="makeStyles-form-293"
        novalidate=""
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled Mui-focused Mui-focused Mui-required Mui-required"
            data-shrink="true"
          >
            Mobile Number
            <span
              aria-hidden="true"
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth Mui-focused Mui-focused MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              autocomplete="username"
              class="MuiInputBase-input MuiOutlinedInput-input"
              name="username"
              required=""
              type="text"
              value="8888888888"
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-407 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-409 PrivateNotchedOutline-legendNotched-410"
              >
                <span>
                  Mobile Number *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled Mui-required Mui-required"
            data-shrink="true"
          >
            Password
            <span
              aria-hidden="true"
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
          >
            <input
              aria-invalid="false"
              autocomplete="current-password"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
              name="password"
              required=""
              type="password"
              value="Abcd@123"
            />
            <button
              aria-label="toggle password visibility"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78l3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-407 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-409 PrivateNotchedOutline-legendNotched-410"
              >
                <span>
                  Password *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <label
          class="MuiFormControlLabel-root Mui-disabled"
        >
          <span
            aria-disabled="true"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-435 MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-disabled-437 Mui-disabled MuiIconButton-colorPrimary Mui-disabled Mui-disabled"
            tabindex="-1"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                class="PrivateSwitchBase-input-438"
                data-indeterminate="false"
                disabled=""
                type="checkbox"
                value="remember"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label Mui-disabled MuiTypography-body1"
          >
            Remember me
          </span>
        </label>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-294 MuiButton-containedPrimary MuiButton-fullWidth"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Sign In
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <div
          class="MuiGrid-root MuiGrid-container"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
          >
            <forget-password />
          </div>
        </div>
      </form>
    </div>
  </main>
</DocumentFragment>
`;

exports[`Login HTML Structure should render properly when showPassword is true 1`] = `
<DocumentFragment>
  <main
    class="MuiContainer-root MuiContainer-maxWidthXs"
  >
    <div
      class="makeStyles-paper-1"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-2 MuiAvatar-colorDefault"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiAvatar-fallback"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
          />
        </svg>
      </div>
      <h1
        class="MuiTypography-root MuiTypography-h5"
      >
        Sign in
      </h1>
      <form
        class="makeStyles-form-3"
        novalidate=""
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled Mui-focused Mui-focused Mui-required Mui-required"
            data-shrink="true"
          >
            Mobile Number
            <span
              aria-hidden="true"
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth Mui-focused Mui-focused MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              autocomplete="username"
              class="MuiInputBase-input MuiOutlinedInput-input"
              name="username"
              required=""
              type="text"
              value="8888888888"
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-117 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-119 PrivateNotchedOutline-legendNotched-120"
              >
                <span>
                  Mobile Number *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled Mui-required Mui-required"
            data-shrink="true"
          >
            Password
            <span
              aria-hidden="true"
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
          >
            <input
              aria-invalid="false"
              autocomplete="current-password"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
              name="password"
              required=""
              type="text"
              value="Abcd@123"
            />
            <button
              aria-label="toggle password visibility"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-117 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-119 PrivateNotchedOutline-legendNotched-120"
              >
                <span>
                  Password *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <label
          class="MuiFormControlLabel-root Mui-disabled"
        >
          <span
            aria-disabled="true"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-145 MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-disabled-147 Mui-disabled MuiIconButton-colorPrimary Mui-disabled Mui-disabled"
            tabindex="-1"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                class="PrivateSwitchBase-input-148"
                data-indeterminate="false"
                disabled=""
                type="checkbox"
                value="remember"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label Mui-disabled MuiTypography-body1"
          >
            Remember me
          </span>
        </label>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-4 MuiButton-containedPrimary MuiButton-fullWidth"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Sign In
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <div
          class="MuiGrid-root MuiGrid-container"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
          >
            <forget-password />
          </div>
        </div>
      </form>
    </div>
  </main>
</DocumentFragment>
`;
